### PR TITLE
Anonymous profiles

### PR DESCRIPTION
**Pull request by Davidjwkinghorn**

### CODE REVIEW

File: src/api/handlers/deviceprofile.py [hunk 1]
```diff
@@ -113,6 +115,75 @@ def log_device(self, request):
...
+    args: dict = context.args
+
+    self.validator.expect("metric", str, is_required=True)
+    self.validator.expect("community_id", str)
+    args, err = self.validator.verify(args)
+
+    if err:
+      return err
+
+    metric = args["metric"] if "metric" in args else None
```
Comment (BradHN1) [inline src/api/handlers/deviceprofile.py:131]: you can do this with args.get("metric", None)

**Review by BradHN1 — APPROVED** [review]
Looks good

File: src/api/handlers/deviceprofile.py [hunk 2]
```diff
@@ -94,7 +96,7 @@ def log_device(self, request):
     if err:
       return err
 
-    self.__get_client_meta(request, args)
+    # self.__get_client_meta(request, args)
```
Comment (Opoku-Agyemang) [inline src/api/handlers/deviceprofile.py:99]: remove all commented out code if not needed.

If you wish to leave any in there, leave a comment on why you prefer to keep the deadcode

**Review by Opoku-Agyemang — CHANGES_REQUESTED** [review]

File: src/api/handlers/deviceprofile.py [hunk 3]
```diff
@@ -113,6 +115,75 @@ def log_device(self, request):
...
+        
+    return MassenergizeResponse(data=metric)
+  
+  @admins_only
+  def community_metrics(self, request):
+    context: Context = request.context
+    args: dict = context.args
+
+    self.validator.expect("metric", str, is_required=True)
+    self.validator.expect("community_id", str)
```
Comment (BradHN1) [inline src/api/handlers/deviceprofile.py:162]: This route looks like it would pass junk back if community_id not specified.  Should it be required?

Comment (BradHN1) [reply src/api/handlers/deviceprofile.py:162]: and if the metric is something different than the three choices, should it return an error?

**Review by BradHN1 — APPROVED** [review]
Looks good - work in progress

File: src/api/handlers/deviceprofile.py (hunk 2 above)
Comment (Davidjwkinghorn) [reply src/api/handlers/deviceprofile.py:99]: I had meant to get round to that. I must have missed a few of those. I'll go back and check.

**Review by BradHN1 — APPROVED** [review]
Very good

**Pull request merged**